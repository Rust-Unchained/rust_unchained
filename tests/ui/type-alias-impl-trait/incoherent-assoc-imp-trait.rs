// Regression test for issue 67856

#![feature(unboxed_closures)]
#![feature(impl_trait_in_assoc_type)]
#![feature(fn_traits)]

trait MyTrait {}
impl MyTrait for () {}

impl<F> FnOnce<()> for &F {
    //~^ ERROR E0119
    type Output = impl MyTrait;
    extern "rust-call" fn call_once(self, _: ()) -> Self::Output {}
}
fn main() {}
