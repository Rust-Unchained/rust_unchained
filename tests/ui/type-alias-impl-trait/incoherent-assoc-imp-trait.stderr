error[E0119]: conflicting implementations of trait `FnOnce()` for type `&_`
  --> $DIR/incoherent-assoc-imp-trait.rs:10:1
   |
LL | impl<F> FnOnce<()> for &F {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<A, F> FnOnce<A> for &F
             where A: Tuple, F: Fn<A>, F: ?Sized;

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
